extensions/wikia/PortableInfobox/tests/PortableInfoboxMobileRenderServiceTest.php

```diff
@@ -0,0 +1,625 @@
+<?php
+
+class PortableInfoboxMobileRenderServiceTest extends WikiaBaseTest {
+	//todo: https://wikia-inc.atlassian.net/browse/DAT-3076
+	//todo: we are testing a lot of functionality and have issues with mocking
+	//todo: we should move all render service test to API tests
+
+	protected function setUp() {
+		$this->setupFile = dirname( __FILE__ ) . '/../PortableInfobox.setup.php';
+		parent::setUp();
+	}
+
+	/**
+	 * @param $input to check presence of some additional config fields. Possible fields:
+	 * 'isInvalidImage' - bool - if getThumbnail should return false
+	 * 'isMercury' - bool - if we want to test Mercury skin
+	 * 'smallImageDimensions' - integer - size of small image (both width and height)
+	 *
+	 * @return PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function mockInfoboxRenderServiceHelper( $input ) {
+		$isValidHeroDataItem = isset( $input['isValidHeroDataItem'] ) && $input['isValidHeroDataItem'];
+		$isMercury = isset( $input['isMercury'] ) && $input['isMercury'];
+
+		$extendImageData = isset( $input['extendImageData'] ) ? $input['extendImageData'] : null;
+
+		$mock = $this->getMockBuilder( 'Wikia\PortableInfobox\Helpers\PortableInfoboxRenderServiceHelper' )
+			->setMethods( [ 'isValidHeroDataItem', 'isMercury', 'extendImageData' ] )
+			->getMock();
+		$mock->expects( $this->any() )
+			->method( 'isValidHeroDataItem' )
+			->will( $this->returnValue( $isValidHeroDataItem ) );
+		$mock->expects( $this->any() )
+			->method( 'isMercury' )
+			->will( $this->returnValue( $isMercury ) );
+		$mock->expects( $this->any() )
+			->method( 'extendImageData' )
+			->will( $this->returnValue( $extendImageData ) );
+
+		$this->mockClass( 'Wikia\PortableInfobox\Helpers\PortableInfoboxRenderServiceHelper', $mock );
+	}
+
+	/**
+	 * @param $html
+	 * @return string
+	 */
+	private function normalizeHTML( $html ) {
+		if ( empty( $hrml ) ) {
+			return '';
+		}
+
+		$DOM = new DOMDocument( '1.0' );
+		$DOM->formatOutput = true;
+		$DOM->preserveWhiteSpace = false;
+		$DOM->loadXML( $html );
+
+		return $DOM->saveXML();
+	}
+
+	/**
+	 * @param $input
+	 * @param $expectedOutput
+	 * @param $description
+	 * @param $mockParams
+	 * @dataProvider testRenderInfoboxDataProvider
+	 */
+	public function testRenderInfobox( $input, $expectedOutput, $description, $mockParams ) {
+		$this->mockInfoboxRenderServiceHelper( $mockParams );
+
+		$infoboxRenderService = new PortableInfoboxMobileRenderService();
+		$actualOutput = $infoboxRenderService->renderInfobox( $input, '', '', '', '' );
+		$expectedHtml = $this->normalizeHTML( $expectedOutput );
+		$actualHtml = $this->normalizeHTML( $actualOutput );
+
+		$this->assertEquals( $expectedHtml, $actualHtml, $description );
+	}
+
+	public function testRenderInfoboxDataProvider() {
+		return [
+			[
+				'input' => [ ],
+				'output' => '',
+				'description' => 'Empty data should yield no infobox markup',
+				'mockParams' => [ ]
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "mockParams" 15 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)